homeassistant/components/media_player/lgtv.py

```diff
@@ -0,0 +1,315 @@
+                sorted_sources = sorted(source_tuples,
+                                        key=lambda channel: int(channel[1]))
+                self._source_names = [n for n, k in sorted_sources]
+        except OSError:
+            self._state = STATE_OFF
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return self._muted
+
+    @property
+    def volume_level(self):
+        """Volume level of the media player (0..1)."""
+        return self._volume / 100.0
+
+    @property
+    def source(self):
+        """Return the current input source."""
+        return self._channel_name
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self._source_names
+
+    @property
+    def media_content_type(self):
+        """Content type of current playing media."""
+        return MEDIA_TYPE_CHANNEL
+
+    @property
+    def media_channel(self):
+        """Channel currently playing."""
+        return self._channel_name
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self._program_name
+
+    @property
+    def supported_media_commands(self):
+        """Flag of media commands that are supported."""
+        return SUPPORT_LGTV
+
+    def turn_off(self):
+        """Turn off media player."""
+        self.send_command(LG_CMD_POWER)
+
+    def volume_up(self):
+        """Volume up the media player."""
+        self.send_command(LG_CMD_VOL_UP)
+
+    def volume_down(self):
+        """Volume down media player."""
+        self.send_command(LG_CMD_VOL_DOWN)
+
+    def mute_volume(self, mute):
+        """Send mute command."""
+        self.send_command(LG_CMD_VOL_MUTE)
+
+    def select_source(self, source):
+        """Select input source."""
+        self._client.change_channel(self._sources[source])
+
+    def media_play_pause(self):
+        """Simulate play pause media player."""
+        if self._playing:
+            self.media_pause()
+        else:
+            self.media_play()
+
+    def media_play(self):
+        """Send play command."""
+        self._playing = True
+        self._state = STATE_PLAYING
+        self.send_command(LG_CMD_PLAY)
+
+    def media_pause(self):
+        """Send media pause command to media player."""
+        self._playing = False
+        self._state = STATE_PAUSED
+        self.send_command(LG_CMD_PAUSE)
+
+    def media_next_track(self):
+        """Send next track command."""
+        self.send_command(LG_CMD_FAST_FORWARD)
+
+    def media_previous_track(self):
+        """Send the previous track command."""
+        self.send_command(LG_CMD_REWIND)
+
+
+# LG TV remote control commands
```

Comment:
Could you extract this class to control the LG TV into a separate Python package that we can install by adding it to our requirements. [Example how it should look](https://github.com/LinuxChristian/pyW215)

You don't have to push it to PyPi, you can just link a specific commit in the `REQUIREMENTS` like [this](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/switch/dlink.py#L18-L19)